app/components/items/item.tsx

```diff
@@ -0,0 +1,68 @@
+import { getItemData } from "@/utils/utils";
+import { ItemModel } from "@/models/item";
+import ImageViewer from "./clover/image/image";
+import CloverImageViewer from "./clover/image/viewer";
+import AudioViewer from "./clover/audio/viewer";
+import VideoViewer from "./clover/video/viewer";
+import BookViewer from "./clover/book/viewer";
+import PDFViewer from "./clover/pdf/viewer";
+
+const getItem = async (uuid) => {
+  const item = await getItemData(uuid);
+  return item;
+};
+
+const Item = ({ item }) => {
+  // const data = await getItem(uuid);
+  // const item = new ItemModel(data);
+
+  const viewer = () => {
```

Comment:
This doesn't need to be a function. If it's a `let` variable, in each `case` you can do something such as
```
viewer  = (<> .... </>)
return;
````
(just because cases should return even undefined). And then on line 65 `<>{viewer}</>` so you don't have to invoke a function call.

Reply:
This is also partly related to the React component life cycle and on every rerender, the value will be assigned and rendered so you don't have to make a function call.

Reply:
I'm having trouble rendering the component with: 

`viewer = (<> .... </>)
 return;`
 
 but if I do this in each case, the component renders and I can remove line 65
 
`viewer = (<> .... </>)
 return viewer;`